mmf/models/m4c.py

```diff
@@ -44,6 +45,19 @@ def build(self):
         self._build_mmt()
         self._build_output()
 
+    def _build_encoder_config(self):
```

Comment:
May I suggest that we move this default config class `_build_encoder_config` to be within the mmf.utils.build class? It is the same with the one in this PR: #564. Perhaps we could add in arguments to the original `build_image_encoder` so that it can be configurable. If the arguments don't exist, we can use those defaults. :)

Reply:
Hey, 
1) This is change is present here because this was exported from stacked diff. #564 is the first diff in that stack and this change is part of it.
2) This is usually not the default image encoder everywhere. So, I would refrain from adding this as default for now. Instead, identity would suit better as default image encoder. It would be better to fail with an error that config was not passed, instead of initializing with something that user doesn't expect by default, IMO.